src/App.jsx

```diff
@@ -1,35 +1,106 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import viteLogo from '/vite.svg'
-import './App.css'
+import React, { useState } from 'react';
 
-function App() {
-  const [count, setCount] = useState(0)
+function App(){
+  const [firstName, setFirstName] = useState('');
+  const [lastName, setLastName] = useState('');
+  const [email, setEmail] = useState('');
+  const [errors, setErrors] = useState({});
+  const [showForm, setShowForm] = useState(false);
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    const errors = {};
+
+    //first name validation
+    if (!firstName) {
```

Comment:
I'd recommend creating a separate function for handling each input field. This way, you only run the validation code you need. This approach would also help with adding the "live validation" requirement which was missed.